src/main/java/com/artozersky/HackerNewsAPI/controller/PostController.java

```diff
@@ -70,24 +65,20 @@ public ResponseEntity<PostResponseDTO> getPost(@PathVariable("id") Long id) {
     @Override
     public ResponseEntity<List<PostResponseDTO>> getAllSortedPosts() {
 
-        List<NewsPostModel> posts = postService.getSortedPostsByScore();
-
-        List<PostResponseDTO> postResponseDTOs = posts.stream()
-                                                    .map(post -> modelMapper.map(post, PostResponseDTO.class))
-                                                    .toList();
+        List<PostResponseDTO> posts = postService.getSortedPostsByScore();
 
-        if (postResponseDTOs.isEmpty()) {
+        if (posts.isEmpty()) {
```

Comment:
Why adding an if? you can return an empty list if it is empty, if no posts the user expects to get empty list and not server error.

Reply:
return empty paginated page please. don't add if.